docs/proposals/improved-schema-for-services.md

```diff
@@ -0,0 +1,562 @@
+              ],
+              "properties": {
+                "containerName": {
+                  "type": "string"
+                },
+                "divisor": {
+                  "type": "string"
+                },
+                "resource": {
+                  "type": "string"
+                }
+              }
+            },
+            "secretKeyRef": {
+              "type": "object",
+              "required": [
+                "key"
+              ],
+              "properties": {
+                "key": {
+                  "type": "string"
+                },
+                "name": {
+                  "type": "string"
+                },
+                "optional": {
+                  "type": "boolean"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "detectBindingResources": {
+    "type": "boolean"
+  },
+  "envVarPrefix": {
+    "type": "string"
+  },
+  "mountPathPrefix": {
+    "type": "string"
+  },
+  "services": {
+    "type": "array",
+    "items": {
+      "type": "object",
+      "required": [
+        "group",
+        "kind",
+        "version"
+      ],
+      "properties": {
+        "envVarPrefix": {
+          "type": "string"
+        },
+        "group": {
+          "type": "string"
+        },
+        "id": {
+          "type": "string"
+        },
+        "kind": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "namespace": {
+          "type": "string"
+        },
+        "version": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}
+
+```
+
+Note - removed `description` fields to make the sample concise. 
+</details>
+
+
+- There are 5 top level fields in the metadata -
+    - `application`
+    - `customEnvVar`
+    - `detectBindingResources`
+    - `envVarPrefix`
+    - `mountPathPrefix`
+    - `services`
+
+### Conversion of openAPIV3Schema to json schema
+
+We after discussions with adapters team have decided to convert the openAPIV3Schema to json schema so that it would be easy to consume and hence we would be working on the converted json schema version to derive information from it.
+
+TODO: find a good enough library to convert openAPIV3Schema to json schema reliably 
+
+### cobra flags to golang map conversion
+
+We need to finalise a mechanism/syntax so that a user can set any property for a CR using the command line, for e.g. if we considered `jsonpath` something like this `odo service create servicebinding.coreos.io/Servicebinding -services[0].envVarPrefix "SVC" -services[0].namespace "openshift"`. 
+
+Currently the decision is to use cobra's syntax for passing flags but we need to finalise and consider all if cobra's syntax will support all scenerios.
+
+### "odo catalog list services"
+
+- No changes needed to `odo catalog list services` as it already shows the `Operators` and the respective CRs they provide for the user to `describe` on.
```

Comment:
For CRDs that have multiple versions, the output should list those versions.

Reply:
added